main.go

```diff
@@ -258,7 +258,15 @@ func main() {
 	if profilerAddress != "" {
 		setupLog.Info("Profiler listening for requests", "profiler-address", profilerAddress)
 		go func() {
-			setupLog.Error(http.ListenAndServe(profilerAddress, nil), "listen and serve error")
+			server := &http.Server{
+				Addr:              profilerAddress,
+				ReadHeaderTimeout: 3 * time.Second,
```

Comment:
From the [docs](https://pkg.go.dev/net/http#Server):

> 	// ReadHeaderTimeout is the amount of time allowed to read
	// request headers. The connection's read deadline is reset
	// after reading the headers and the Handler can decide what
	// is considered too slow for the body. If ReadHeaderTimeout
	// is zero, the value of ReadTimeout is used. If both are
	// zero, there is no timeout.

So I'm guessing that means with the previous code, this value was effectively 0, meaning "no timeout"?

3 seconds seems generous to read an HTTP header, so I think this is probably a safe change.